zebra-state/src/lib.rs

```diff
@@ -93,3 +108,11 @@ pub enum Response {
         Option<u32>,
     ),
 }
+
+/// Get the heights of the blocks for constructing a block_locator list
+fn block_locator_heights(tip_height: BlockHeight) -> impl Iterator<Item = BlockHeight> {
+    iter::successors(Some(1u32), |h| h.checked_mul(2))
+        .flat_map(move |step| tip_height.0.checked_sub(step))
+        .map(BlockHeight)
+        .chain(iter::once(BlockHeight(0)))
```

Comment:
I really like how simple this iterator chaining is.

Reply:
sameeee